core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -51,35 +51,57 @@ static std::atomic<bool> is_first_hip_managed_allocation(true);
 
 namespace Kokkos {
 
-HIPSpace::HIPSpace() : m_device(HIP().hip_device()) {}
+HIPSpace::HIPSpace()
+    : m_device(HIP().hip_device()), m_stream(Kokkos::HIP().hip_stream()) {}
```

Comment:
```suggestion
HIPSpace::HIPSpace(const HIP& exec)
    : m_device(exec.hip_device()), m_stream(exec.hip_stream()) {}
```

Reply:
(and probably `explicit`)

Reply:
No

Reply:
You still have mismatch between `HIP().hip_device()` and `Kokkos::HIP().hip_stream()`...

Reply:
I don't understand what you mean with "there is a mismatch". The extra namespace?

Reply:
yes